File: backend/src/main/java/it/unical/inf/ea/backend/dto/ProductDTO.java
package it.unical.inf.ea.backend.dto;

import com.fasterxml.jackson.annotation.JsonTypeInfo;
import it.unical.inf.ea.backend.data.entities.ProductImage;
import it.unical.inf.ea.backend.dto.enums.Availability;
import jakarta.validation.constraints.NotNull;
import lombok.*;
import org.hibernate.validator.constraints.Length;

import java.math.BigDecimal;
import java.util.List;

@Data
@NoArgsConstructor
@ToString
@AllArgsConstructor
@Builder
public class ProductDTO {

    @NotNull
    private String id;

    @Length(max = 100)
    @NotNull
    private String title;

    @Length(max = 1000)
    private String description;

    @Length(max = 1000)
    private String ingredients;

    @Length(max = 1000)
    private String nutritionalValues;

    @NotNull
    private BigDecimal productPrice;

    @NotNull
    private BigDecimal deliveryPrice;

    @NotNull
    private BrandDTO brand;

    private String productWeight;

    @NotNull
    private int quantity;

    @NotNull
    private Availability availability;

    @NotNull
    private ProductCategoryDTO productCategory;

    private ProductImageDTO photoProduct;

    @NotNull
    private boolean onSale;

    private BigDecimal discountedPrice;

}

File: frontend/app/src/main/java/com/android/frontend/controller/models/ProductDTO.kt
package com.android.frontend.controller.models

import java.math.BigDecimal

data class ProductDTO (

    val id: String,
    val title: String,
    val description: String,
    val ingredients:String,
    val nutritionalValues: String,
    val productPrice: BigDecimal,
    val deliveryPrice: BigDecimal,
    val brand:BrandDTO,
    val productWeight: String,
    val quantity: Int,
    val availability: Availability,
    val productCategory: ProductCategoryDTO,
    val productImages: List<ProductImageDTO>,
) {


    enum class Availability(val value:String){
        AVAILABLE("AVAILABLE"),
        PENDING("PENDING"),
        UNAVAILABLE("UNAVAILABLE");
    }
}